main/lsp/LSPTypechecker.h

```diff
@@ -55,6 +55,9 @@ class LSPTypechecker final {
     UnorderedMap<int, ast::ParsedFile> indexedFinalGS;
     /** Hashes of global states obtained by resolving every file in isolation. Used for fastpath. */
     std::vector<core::FileHash> globalStateHashes;
+    /** Stores the epoch in which we last sent diagnostics to the client for each file. Should be the same length as
```

Comment:
Thought: did you consider combining `globalStateHashes` and `diagnosticEpochs` into a single vector of `FileHash`/`u4` pairs so that it would be impossible for them to be of different lengths?  (Or is it important to store them separately to save memory?)

Reply:
Good point! I have them separate because it was convenient given the internal Sorbet APIs I was working with. For example, the various typechecking passes take a vector of `ast::ParsedFile`.

However, I don't think we're passing any of these directly to any of those methods, so it might make sense to combine `indexed` / `globalStateHashes` / `diagnosticEpochs`. I'll add it to a list of things to consider in a future PR.